src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -1127,7 +1161,7 @@ function setupArticleTocSyncing() {
     }
 
     observer = new IntersectionObserver(callback, options);
-    headingsToTocLinks.keys().forEach((heading) => {
+    Array.from(headingsToTocLinks.keys()).forEach((heading) => {
```

Comment:
This was actually causing a runtime error in Safari for me. Only the latest versions of Safari support forEach on the [map iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator).